src/nexgen/tools/ED_tools.py

```diff
@@ -173,6 +184,7 @@ def extract_from_SINGLA_master(master: Path | str) -> Dict[str, Any]:
         D["detector_number"] = singla.get_detector_number()
         D["detector_readout_time"] = singla.get_detector_readout_time()
         D["photon_energy"] = singla.get_photon_energy()
+        D["data_collection_date"] = singla.get_data_collection_date()
```

Comment:
Must: move this bit to its own function, also in ED_tools, to be called from the ED_nexus script. 

Reason is that, while convenient, unfortunately having this added to the Detector dictionary may cause the script to write a "data_collection_data" dataset in the wrong place in the NeXus tree. This will make the file non-standard.